Tools/Scripts/libraries/webkitcorepy/webkitcorepy/file_lock.py

```diff
@@ -22,60 +22,86 @@
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import os
+import re
 import sys
 import time
 
+from webkitcorepy import string_utils
+
 if sys.platform.startswith('win'):
     import msvcrt
-else:
-    import fcntl
 
 
 class FileLock(object):
+    INTEGER_RE = re.compile(r'^\d+$')
+    USE_WINDOWS = sys.platform.startswith('win')
+    USE_EXLOCK = not USE_WINDOWS and getattr(os, 'O_EXLOCK', False)
+
+    @classmethod
+    def is_process_running(cls, pid):
+        try:
+            os.kill(pid, 0)
```

Comment:
Seems worth asserting that this isn't getting called on Windows. The current code doesn't, but that's fragile.